js/stores/windowStore.js

```diff
@@ -135,14 +135,10 @@ const doAction = (action) => {
     case WindowConstants.WINDOW_SET_LOCATION:
       const key = action.key || windowState.get('activeFrameKey')
       const lastLocation = windowState.getIn(frameStatePath(key).concat(['location']))
-      const lastTitle = windowState.getIn(frameStatePath(key).concat(['title']))
-      let locationChanged = !action.location || !lastLocation ||
-        action.location.split('#')[0] !== lastLocation.split('#')[0]
       windowState = windowState.mergeIn(frameStatePath(key), {
         audioPlaybackActive: false,
         adblock: {},
         trackingProtection: {},
-        title: locationChanged ? '' : lastTitle,
```

Comment:
I'm fairly sure that doing this will break when you navigate to a page with a title and then after that to a page with no title.  This area of code is desperately in need of tests.